src/Loader/DirectoryLoader.php

```diff
@@ -59,6 +61,7 @@ public function supports(mixed $resource)
     public function load(mixed $resource)
     {
         $path = $this->getAbsolutePath($resource);
+        /** @var Traversable<string|SplFileInfo> $iterator */
```

Comment:
(PHPStan) PHPStan doesn't infer the iterator correctly (I think there's a bug/feature issue about this somewhere?).

Reply:
why `string|SplFileInfo` ? As we use the default flags, it should be `SplFileInfo` only AFAICT.

anyway, this indeed looks like a case where the phpstan inference is not good.